File: 2024/template.py
import re
import time
from argparse import ArgumentParser

def read(file):
  with open(file, "r") as f:
    for line in f:
      pass


def main(file):
  pass


if __name__ == "__main__":
  parser = ArgumentParser()
  parser.add_argument("file")
  args = parser.parse_args()
  start_time = time.time()
  print("\nRESULT:", main(args.file))
  print("--- COMPLETED IN %s SECONDS ---" % (time.time() - start_time))

File: 2024/template.go
package main

import (
	"bufio"
	"fmt"
	"os"
	"time"
)

func read(filename string) {
	file, err := os.Open(filename)
	if err != nil {
		panic(err)
	}
	defer file.Close()
	scanner := bufio.NewScanner(file)
	for scanner.Scan() {
		line := scanner.Text()
		fmt.Println(line)
	}
}

func run(filename string) int {
	read(filename)
	total := 0
	return total
}

func main() {
	startTime := time.Now()
	filename := os.Args[1]
	fmt.Println("RESULT:", run(filename))
	elapsed := time.Since(startTime)
	fmt.Println("completed in", elapsed.Seconds(), "seconds")
}
